hw02-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,32 @@
+package homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private TreeMap<Customer, String> customers = new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```

Comment:
```suggestion
    private NavigableMap<Customer, String> customers = new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```

Тип лучше объявлять интерфейсом, причём минимально необходимым. Здесь будет достаточно `NavigableMap`.